scvid/module/_probabilisticpca.py

```diff
@@ -0,0 +1,175 @@
+import pyro
+import pyro.distributions as dist
+import torch
+from pyro.nn import PyroModule, PyroParam
+from scvi.module.base import PyroBaseModuleClass
+from torch.distributions import constraints
+
+_PROBABILISTIC_PCA_PYRO_MODULE_NAME = "probabilistic_pca"
+
+
+class ProbabilisticPCAPyroModel(PyroModule):
+    """
+    A PyroModule that serves as the model for the ProbabilisticPCAPyroModule class.
+
+    Args:
+        n_obs: Number of observations.
+        n_vars: Number of input features.
+        n_components: Number of components to model.
+        mean: Mean of the input data.
+        marginalize_z: Marginalize out latent variable z.
+    """
+
+    def __init__(
+        self,
+        n_obs: int,
+        n_vars: int,
+        n_components: int,
+        mean: torch.Tensor,
+        marginalize_z: bool,
+    ):
+        super().__init__(_PROBABILISTIC_PCA_PYRO_MODULE_NAME)
+
+        self.n_obs = n_obs
+        self.n_vars = n_vars
+        self.n_components = n_components
+        self.mean = mean
+        self.marginalize_z = marginalize_z
+
+        # model parameters
+        self.W = PyroParam(lambda: torch.randn((n_components, n_vars)))
```

Comment:
Please let the scale of the random initialization of `W` (and every other model parameter) to be user-specified. There are often good heuristics to guess the _magnitude_ of various model params, and so you'd want to expose these to the user. For example, let's say we know the empirical variance of the dataset, and let that be `s^2`. A PPCA model with Gaussian initialization of `W` witth scale `w` produces the following variance:
```
E[tr(W^T W) + F \sigma^2] = F D w^2 + F \sigma^2 ~ s^2
```
where `F` is the ambient dimension = number of features, and we have `D` PCs. Now, let's say we expect to be able to explain `90%` of the variance with the PCs and leave `10%` for the isotropic noise. The we have:
```
\sigma^2 ~ 0.1 s^2 / F
w^2 ~ 0.9 s^2 / (F x D)
```